bread/layout/components/tooltip.py

```diff
@@ -0,0 +1,268 @@
+                        "bx--tooltip--a11y "
+                        "bx--tooltip__trigger--definition "
+                        "bx--tooltip--%s "
+                        "bx--tooltip--align-%s"
+                    )
+                    % (position, align),
+                    aria_describedby=asst_txt_id,
+                ),
+                hg.DIV(
+                    description,
+                    _class="bx--assistive-text",
+                    id=asst_txt_id,
+                    role="tooltip",
+                ),
+                **tooltip_attributes,
+            )
+        )
+
+
+class IconTooltip(hg.BaseElement):
+    """Icon tooltip is for short single line of text describing an icon. Icon tooltip
+    does not use any JavaScript. No label should be added to this variation. If there
+    are actions a user can take in the tooltip (e.g. a link or a button),
+    use interactive tooltip.
+
+    Reference: https://the-carbon-components.netlify.app/?nav=tooltip
+    """
+
+    def __init__(
+        self,
+        description: str,
+        icon: Union[Icon, str] = "information",
+        align: str = "center",
+        position: str = "bottom",
+        **attributes,
+    ):
+        """
+        Parameters
+        ----------
+        description : str
+            One line of string describing an icon.
+        icon : Icon, str, optional
+            Specify an icon for the tooltip.
+            The default value is 'information'.
+        align : str, optional
+            Specify where the arrow pointing the icon should align to the text.
+            It can be either start, center, or end.
+            The default value is 'center'.
+        position : str
+            Where should the tooltip appear besides the tooltip icon.
+            It can be either top, left, right, or bottom.
+            The default value is 'bottom'.
+        """
+        tooltip_attributes = {
+            "_class": (
+                "bx--tooltip__trigger "
+                "bx--tooltip--a11y "
+                "bx--tooltip--%s "
+                "bx--tooltip--align-%s"
+            )
+            % (position, align),
+            "data_tooltip_icon": True,
+        }
+
+        tooltip_attributes = hg.merge_html_attrs(tooltip_attributes, attributes)
+        icon = _get_icon(icon)
+
+        super().__init__(
+            hg.BUTTON(
+                hg.SPAN(description, _class="bx--assistive-text"),
+                icon,
+                **tooltip_attributes,
+            )
+        )
+
+
+class InteractiveTooltip(hg.BaseElement):
+    """Interactive tooltip should be used if there are actions a user can take in
+    the tooltip (e.g. a link or a button). For more regular use case, e.g. giving
+    the user more text information about something, use definition tooltip or icon
+    tooltip.
+
+    Reference: https://the-carbon-components.netlify.app/?nav=tooltip
+    """
+
+    base_class = "bx--tooltip"
```

Comment:
I would move these into the constructor because having them here suggests that you are using them in a subclass or sharing them between multiple methods, which you are not doing currently.

Reply:
Hmm... I think so. Thank you.